xarray/plot/facetgrid.py

```diff
@@ -8,7 +8,7 @@
 from .utils import (
     _infer_xy_labels, _process_cmap_cbar_kwargs, import_matplotlib_pyplot,
     label_from_attrs)
-
+import xarray as xr
```

Comment:
Is this really needed?

Reply:
I needed it for Line 341, but I see your point. I replaced it with 

`from ..core.dataarray import DataArray`

Is that better?

Reply:
I needed it for checking the legend but I found a different workaround:
https://github.com/pydata/xarray/blob/21353888a38e777bf59c397a7dcaa5a20c132668/xarray/plot/facetgrid.py#L340

Reply:
Can we just make sure we either only pass DataArrays or their underlying numpy arrays to `add_legend`?

Reply:
OK. I'll try to look into it over the weekend